Framework/DataHandling/src/SaveCanSAS1D.cpp

```diff
@@ -17,6 +17,36 @@
 #include <boost/shared_ptr.hpp>
 
 //-----------------------------------------------------------------------------
+namespace {
+void encode(std::string &data) {
+  std::string buffer;
+  buffer.reserve(data.size());
+  for (size_t pos = 0; pos != data.size(); ++pos) {
+    switch (data[pos]) {
+    case '&':
+      buffer.append("&amp;");
+      break;
+    case '\"':
+      buffer.append("&quot;");
+      break;
+    case '\'':
+      buffer.append("&apos;");
+      break;
+    case '<':
+      buffer.append("&lt;");
+      break;
+    case '>':
+      buffer.append("&gt;");
+      break;
+    default:
+      buffer.append(&data[pos], 1);
+      break;
+    }
+  }
```

Comment:
It's not that important but I would write the loop like this:

```
for(auto c : data) {
  switch(c) {
    ...
  default:
    buffer.append(1, c);  // or buffer.push_back(c);
  }
}
```

This would avoid unnecessary indexing operations.